spectrum/checks.py

```diff
@@ -491,17 +491,21 @@ class JournalCheck:
     CLASS_FIGURES_LINK = 'view-selector__link--figures'
     CLASS_SUBJECT_LINK = 'content-header__subject_link'
 
-    def __init__(self, host, resource_checking_method='head', query_string=None):
+    def __init__(self, host, resource_checking_method='head', query_string=None, headers=None):
         self._host = host
         self._resource_checking_method = resource_checking_method
         self._query_string = query_string
+        self._headers = headers
 
     def with_resource_checking_method(self, method):
         return JournalCheck(self._host, method, self._query_string)
 
     def with_query_string(self, query_string):
         return JournalCheck(self._host, self._resource_checking_method, query_string)
```

Comment:
this and `with_resource_checking_method` are meant to allow creating a new object while keeping the other settings, so `self._headers` should be added at the end